classes/Link.php

```diff
@@ -921,14 +921,17 @@ public function getAdminBaseLink($idShop = null, $ssl = null, $relativeProtocol
             } else {
                 $shop = new Shop((int) Configuration::get('PS_SHOP_DEFAULT'));
             }
+            list($domainSSL, $domain) = $this->getShopDomainActive($shop, $idShop);
         } else {
             $shop = Context::getContext()->shop;
+            $domainSSL = $shop->domain_ssl;
+            $domain = $shop->domain;
         }
 
         if ($relativeProtocol) {
-            $base = '//' . ($ssl && $this->ssl_enable ? $shop->domain_ssl : $shop->domain);
+            $base = '//' . ($ssl && $this->ssl_enable ? $domainSSL : $domain);
```

Comment:
It looks like there is already a check in Tools::getShopProtocol(), can you please use this ?